packages/vexide-devices/src/adi/digital.rs

```
@@ -163,6 +163,32 @@ impl AdiDigitalOut {
         Self { port }
     }
 
+    /// Create a digital output from an [`AdiPort`] with an initial logic level.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use vexide::prelude::*;
+    ///
+    /// #[vexide::main]
+    /// async fn main(peripherals: Peripherals) {
+    ///     let digital_out = AdiDigitalOut::with_initial_level(peripherals.adi_a, LogicLevel::High).expect("failed to initialize digital output");
+    /// }
+    /// ```
+    ///
+    /// # Errors
+    ///
+    /// - A [`PortError::Disconnected`] error is returned if an ADI expander device was required but not connected.
+    /// - A [`PortError::IncorrectDevice`] error is returned if an ADI expander device was required but
+    ///   something else was connected.
+    pub fn with_initial_level(port: AdiPort, initial_level: LogicLevel) -> Result<Self, PortError> {
```

In general we are against fallible constructors whenever possible because it can eat the SmartPort and is in general difficult to recover from, so it would be better if this is infallible.

---

Should this not exist, then, or silently drop the error? It seems like something that would be useful.

---

The error is not silently dropped; VEXos will save the value and set the port as soon as the expander is plugged back in.